logstash-core-event-java/src/main/java/com/logstash/KeyNode.java

```diff
@@ -39,4 +39,20 @@ public String evaluate(Event event) throws IOException {
             return "%{" + this.key + "}";
         }
     }
+
+    // TODO: (colin) this should be moved somewhere else to make it reusable
+    //   this is a quick fix to compile on JDK7 a not use String.join that is
+    //   only available in JDK8
+    public static String join(List<?> list, String delim) {
```

Comment:
there is an indirect test on it in the StringInterpolationTest, it is definitely not complete as far as specifically testing the `join` method per-se. But yeah, properly testing join is a good idea, and we can move those tests whenever we refactor it.